From 53efb2fc5916f1327d7b55b8ebacfa1f477f664f Mon Sep 17 00:00:00 2001
Message-Id: <53efb2fc5916f1327d7b55b8ebacfa1f477f664f.1432858002.git.chang-joon.lee@intel.com>
In-Reply-To: <944dfa31d45b9ee101932dfa029ec3452a83cb30.1432858002.git.chang-joon.lee@intel.com>
References: <944dfa31d45b9ee101932dfa029ec3452a83cb30.1432858002.git.chang-joon.lee@intel.com>
From: Minu <minu.mathai@intel.com>
Date: Fri, 15 May 2015 14:51:50 +0100
Subject: [PATCH 2/7] FOR_UPSTREAM [VPG]: Correcting the reg definitions for
 PORT_DFT

Display CRCs were not readable because the register defintions
for PORT_DFT_I9XX and PORT_DFT2_G4X were wrong.
MMIO offset needs to be added to these register offsets to fix them.

Issue: GMINL-6869
Change-Id: I73a35a7d70e9046cc5f8af62ce90a680943d1116
Signed-off-by: Minu Mathai <minu.mathai@intel.com>
---
 drivers/gpu/drm/i915/i915_reg.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/gpu/drm/i915/i915_reg.h b/drivers/gpu/drm/i915/i915_reg.h
index 62fd9a5..bc58c6a 100644
--- a/drivers/gpu/drm/i915/i915_reg.h
+++ b/drivers/gpu/drm/i915/i915_reg.h
@@ -2956,9 +2956,9 @@ enum punit_power_well {
 #define _PORTADDR(dvo_port, b, c, d)	((dvo_port == b) ? GEN4_HDMIB : \
 					((dvo_port == c) ? GEN4_HDMIC : \
 					CHV_HDMID))
-#define PORT_DFT_I9XX				0x61150
+#define PORT_DFT_I9XX				(dev_priv->info.display_mmio_offset + 0x61150)
 #define   DC_BALANCE_RESET			(1 << 25)
-#define PORT_DFT2_G4X				0x61154
+#define PORT_DFT2_G4X				(dev_priv->info.display_mmio_offset + 0x61154)
 #define   DC_BALANCE_RESET_VLV			(1 << 31)
 #define   PIPE_SCRAMBLE_RESET_MASK		(0x3 << 0)
 #define   PIPE_B_SCRAMBLE_RESET			(1 << 1)
-- 
1.9.1

